avocado/plugins/nrun.py

```diff
@@ -103,9 +106,38 @@ def spawn_tasks(self):
             self.spawned_tasks.append(identifier)
             print("%s spawned" % identifier)
 
-    @staticmethod
+    def pick_runner(self, task):
```

Comment:
Hi, @clebergnu what about use __subclasses__() of BaseRunner and get all runners for searching for runners. I think it will be better because, when we add new runer the inheritance from BaseRunner will be the only thing to do. What do you think? This is just an idea for the discussion because it means a completely different approach.

Reply:
The idea for the nrunner is really to be implementation agnostic (even with regards to language) and rely only on:

 1)  command line conventions
 2) JSON-like output

This clearly introduces a new convention, which is a Python module naming convention for the "job" to find a suitable runner, but it's coupled way more loosely than trying to find a subclass intention, for the reasons listed above.  So, in theory, a new runner implementation could decide to do all things on its on.

So, while I understand your proposal (which would make sense without the points I listed), it wouldn't work or be desirable here.

Reply:
Thanks for the explanation, I was thinking that this will be easier for the implementation. But you are right that for runners it will be much more binding.